.github/workflows/deployment_daedalusproject.yml

```diff
@@ -0,0 +1,83 @@
+name: CI
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+  pull_request:
+    branches: [main]
+
+  # Allows you to run this workflow manually from the Actions tab.
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python version
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Create and start virtual environment
+        run: |
+          python -m venv venv
+          source venv/bin/activate
+
+      - name: Install dependencies
+        run: |
+          pip install -r requirements.txt
+
+
+      - name: Install dependencies frontend
+        run: |
+          cd web
+          npm install
+
+      - name: Run the frontend tests
+        run: |
+          cd web
+          npm test
+
+      - name: Build
+        run: |
+          cd web
+          npm run build
+          
+      - name: Nuke node_modules
+        run: |
+          cd web
+          rm -rf node_modules
+
+      - name: Upload artifact for deployment jobs

```

Comment:
Can we make it so this only happens on master? 

Is there anyway we could specify certain branches for deployment and have them deploy to different urls based on the branch name?